kv/memberlist/memberlist_client.go

```diff
@@ -565,7 +565,14 @@ func (m *KV) fastJoinMembersOnStartup(ctx context.Context) {
 	l.Log("msg", "memberlist fast-join finished", "joined_nodes", totalJoined, "elapsed_time", time.Since(startTime))
 }
 
+// The joinMembersOnStartup method resolves the addresses of the given join_members hosts and asks memberlist to join to them.
+// This method cannot be called before KV.running state as it may wait for K8S DNS to resolve the service addresses of members
+// running this very method. Which means the service needs to be READY for K8S to add it to DNS.
 func (m *KV) joinMembersOnStartup(ctx context.Context) bool {
+	if len(m.cfg.JoinMembers) == 0 {
+		return true
+	}
+
```

Comment:
We don't need this check, it's already done before calling `joinMembersOnStartup`: https://github.com/grafana/dskit/blob/26c1b619d0599b44104131d1f3a61ddbec63e188/kv/memberlist/memberlist_client.go#L478

Reply:
I'd rather remove the if on the outside, to establish an invariant at the top of the function

Reply:
That works too.